serve: crashes at startup when `CLIENT_AUTH_PUBLIC_KEYS` is empty, although auth is documented as optional

<!-- Please answer these questions before submitting your issue. Thanks! -->

### What version are you using?

6658072 (current `main`)

### What did you do?

Started the server with `CLIENT_AUTH_PUBLIC_KEYS` unset (empty), expecting to run
without client authentication:

```
wallet-backend serve
```

### What happened?

The server panics at startup

```
panic: executing root command: running serve: setting up handler dependencies: instantiating multi JWT token parser: no Stellar public keys provided
```

### What did you expect to see instead?
The server starts with request authentication disabled, as advertised by:

- the `--client-auth-public-keys` flag help: *"If not provided or empty,
  authentication is disabled."* (`cmd/serve.go`, `Required: false`)

--- a/README.md
+++ b/README.md
@@ -171,6 +171,8 @@ The JWT is signed using an Ed25519 private key derived from a Stellar secret see
 
 The server can be configured to accept multiple (comma-separated) public keys through the `CLIENT_AUTH_PUBLIC_KEYS` environment variable.
 
+If `CLIENT_AUTH_PUBLIC_KEYS` is not set or empty, request authentication is disabled.
+
 ### JWT Claims
 
 The JWT payload field should contain the following fields:

--- a/internal/serve/serve.go
+++ b/internal/serve/serve.go
@@ -176,11 +176,10 @@ func initHandlerDeps(ctx context.Context, cfg Configs) (handlerDeps, error) {
 		return handlerDeps{}, fmt.Errorf("creating models for Serve: %w", err)
 	}
 
-	jwtTokenParser, err := auth.NewMultiJWTTokenParser(time.Duration(cfg.ClientAuthMaxTimeoutSeconds)*time.Second, cfg.ClientAuthPublicKeys...)
+	requestAuthVerifier, err := buildRequestAuthVerifier(ctx, cfg)
 	if err != nil {
-		return handlerDeps{}, fmt.Errorf("instantiating multi JWT token parser: %w", err)
+		return handlerDeps{}, err
 	}
-	requestAuthVerifier := auth.NewHTTPRequestVerifier(jwtTokenParser, int64(cfg.ClientAuthMaxBodySizeBytes))
 
 	httpClient := http.Client{Timeout: 30 * time.Second}
 	rpcService, err := services.NewRPCService(cfg.RPCURL, cfg.NetworkPassphrase, &httpClient, m.RPC)
@@ -204,6 +203,21 @@ func initHandlerDeps(ctx context.Context, cfg Configs) (handlerDeps, error) {
 		NetworkPassphrase:         cfg.NetworkPassphrase,
 		GraphQLComplexityLimit:    cfg.GraphQLComplexityLimit,
 	}, nil
+}
+
+// buildRequestAuthVerifier returns nil when no client auth public keys are
+// configured, which makes handler() skip the authentication middleware entirely.
+func buildRequestAuthVerifier(ctx context.Context, cfg Configs) (auth.HTTPRequestVerifier, error) {
+	if len(cfg.ClientAuthPublicKeys) == 0 {
+		log.Ctx(ctx).Warn("client-auth-public-keys is empty: request authentication is disabled")
+		return nil, nil
+	}
+
+	jwtTokenParser, err := auth.NewMultiJWTTokenParser(time.Duration(cfg.ClientAuthMaxTimeoutSeconds)*time.Second, cfg.ClientAuthPublicKeys...)
+	if err != nil {
+		return nil, fmt.Errorf("instantiating multi JWT token parser: %w", err)
+	}
+	return auth.NewHTTPRequestVerifier(jwtTokenParser, int64(cfg.ClientAuthMaxBodySizeBytes)), nil
 }
 
 func handler(deps handlerDeps) http.Handler {

--- a/internal/serve/serve_test.go
+++ b/internal/serve/serve_test.go
@@ -1,0 +1,44 @@
+package serve
+
+import (
+	"context"
+	"testing"
+
+	"github.com/stellar/go-stellar-sdk/keypair"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestBuildRequestAuthVerifier(t *testing.T) {
+	baseCfg := Configs{
+		ClientAuthMaxTimeoutSeconds: 15,
+		ClientAuthMaxBodySizeBytes:  102_400,
+	}
+
+	t.Run("auth disabled when no public keys are provided", func(t *testing.T) {
+		cfg := baseCfg
+		cfg.ClientAuthPublicKeys = nil
+
+		verifier, err := buildRequestAuthVerifier(context.Background(), cfg)
+		require.NoError(t, err)
+		assert.Nil(t, verifier)
+	})
+
+	t.Run("auth enabled when public keys are provided", func(t *testing.T) {
+		cfg := baseCfg
+		cfg.ClientAuthPublicKeys = []string{keypair.MustRandom().Address()}
+
+		verifier, err := buildRequestAuthVerifier(context.Background(), cfg)
+		require.NoError(t, err)
+		assert.NotNil(t, verifier)
+	})
+
+	t.Run("invalid public key returns error", func(t *testing.T) {
+		cfg := baseCfg
+		cfg.ClientAuthPublicKeys = []string{"not-a-stellar-public-key"}
+
+		verifier, err := buildRequestAuthVerifier(context.Background(), cfg)
+		require.ErrorContains(t, err, "instantiating multi JWT token parser")
+		assert.Nil(t, verifier)
+	})
+}

